lrauv_ignition_plugins/src/ScienceSensorsSystem.cc

```diff
@@ -933,6 +926,12 @@ float ScienceSensorsSystemPrivate::BarycentricInterpolate(
   // Retrieve indices of sorted elements in original array
   int ltPIdx = xsSortedInds[ltPSortedIdx];
   int gtPIdx = xsSortedInds[gtPSortedIdx];
+
+  if (ltPIdx >= _values.size() || gtPIdx >= _values.size())
```

Comment:
This too should not happen. I'd like to reproduce to see why it happens and check it earlier in the code.

Reply:
This one indeed doesn't happen for me, but while debugging I figured I'd add a check just in case. It may fail in the future...